nanoFramework.Json/JsonConvert.cs

```diff
@@ -1030,7 +1132,24 @@ private static JsonToken ParseValue(ref LexToken token)
                 }
                 else
                 {
-                    return new JsonValue(int.Parse(token.TValue));
+                    try
+                    {
+                        return new JsonValue(int.Parse(token.TValue));
+                    }
+                    catch
+                    {
+                        try
+                        {
+                            return new JsonValue(long.Parse(token.TValue));
+                        }
+                        catch
+                        {
+                            {
```

Comment:
Is the extra set of curly braces voluntary ?

Reply:
good catch, it's not!